processor/src/errors.rs

```diff
@@ -21,10 +28,99 @@ pub enum ExecutionError {
     NttDomainSizeTooSmall(u64),
     NttDomainSizeNotPowerof2(u64),
     InvalidStackDepthOnReturn(usize),
-    InterpolationResultSizeTooBig(u64),
+    InterpolationResultSizeTooBig(usize, usize),
     NotBinaryValue(Felt),
     NotU32Value(Felt),
     ProverError(ProverError),
     SyscallTargetNotInKernel(Digest),
     UnexecutableCodeBlock(CodeBlock),
 }
+
+impl Display for ExecutionError {
+    fn fmt(&self, fmt: &mut Formatter<'_>) -> Result<(), core::fmt::Error> {
+        match self {
+            ExecutionError::AdviceKeyNotFound(key) => {
+                let hex = hex::encode(Felt::elements_as_bytes(key));
+                write!(fmt, "Can't write to advice tape: value for key {hex} not present in the advice map.")
+            }
+            ExecutionError::AdviceSetLookupFailed(reason) => {
+                write!(fmt, "Advice set lookup failed: {reason}")
+            }
+            ExecutionError::AdviceSetNotFound(root) => {
+                write!(fmt, "Advice set with root {root:x?} not found")
+            }
+            ExecutionError::AdviceSetUpdateFailed(reason) => {
+                write!(fmt, "Advice set update failed: {reason}")
+            }
+            ExecutionError::AdviceTapeReadFailed(step) => {
+                write!(fmt, "Advice tape read failed at step {step}")
+            }
+            ExecutionError::CodeBlockNotFound(digest) => {
+                let hex = hex::encode(digest.as_bytes());
+                write!(
+                    fmt,
+                    "Failed to execute code block with root {hex}; the block could not be found"
+                )
+            }
+            ExecutionError::CallerNotInSyscall => {
+                write!(
+                    fmt,
+                    "Instruction `caller` used outside of kernel context, this is not supported"
+                )
+            }
+            ExecutionError::DivideByZero(clk) => {
+                write!(fmt, "Division by zero at clock cycle {clk}")
+            }
+            ExecutionError::DuplicateAdviceKey(key) => {
+                let bytes = Felt::elements_as_bytes(key);
+                write!(fmt, "Insertion into advice map failed because {bytes:x?} already exists")
+            }
+            ExecutionError::FailedAssertion(clk) => {
+                write!(fmt, "Assertion failed at clock cycle {clk}")
+            }
+            ExecutionError::UninitializedMemoryAddress(address) => {
+                write!(fmt, "Ext2INTT referenced unintialized memory at address {address}")
+            }
+            ExecutionError::InvalidFmpValue(old, new) => {
+                write!(fmt, "Updating FMP register failed, from {old} to {new}, because {new} is outside of {FMP_MIN}..{FMP_MAX}")
```

Comment:
minor nit: maybe change the text to: "Updating FMP register from {old} to {new} failed because {new} is outside of {FMP_MIN}..{FMP_MAX}"